utils/backup/S3/fo-backup-s3

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# Copyright (C) Siemens AG 2017
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# version 2 as published by the Free Software Foundation.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source ${DIR}/before
+set -x
+${DIR}/install
+
+echo Starting backup at `date`
+
+
+
+echo Creating diff of repository at `date`
+ls -RlahGg --time-style='+' /srv/fossology/repository/ | cut -d ' ' -f 2-5 | ${DIR}/venv/bin/aws s3 cp --sse AES256 - "s3://${BUCKET}/${FILENAME}-repo-diff.txt"
+
+echo Start database dump at `date`
+#### postgresql needs to be restarted for the dump to work
+/etc/init.d/postgresql restart
+#### these steps are split, because of permissions
+TMP_FILE=`mktemp`
+su postgres -c 'pg_dumpall' > ${TMP_FILE}
+cp ${TMP_FILE} /srv/fossology/database.txt
+echo Done database dump. Starting Upload at `date`
+
+#### sync repository
+echo "Uploading and starting backup to S3..."
+tar -zcvf - /srv/fossology/ | ${DIR}/venv/bin/aws --debug s3 cp --expected-size 200000000000 --sse AES256 - "s3://${BUCKET}/${FILENAME}"
```

Comment:
Yes, but maybe the default value will change overtime and then it wouldn't work anymore. What do you think?